### PR TITLE
Cleanup XMPUtil and PdfImporter

### PR DESCRIPTION
**Pull request by koppor**
This is a minor cleanup on XMPUtil and PdfImporter to ease next development steps

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 1]
```diff
@@ -105,9 +113,11 @@ public static void writeXMP(String filename, BibEntry entry,
      *             than remove a lock or cancel the operation.
      */
     public static List<BibEntry> readXMP(File file) throws IOException {
+        List<BibEntry> res = Collections.EMPTY_LIST;
         try (FileInputStream is = new FileInputStream(file)) {
-            return XMPUtil.readXMP(is);
+            res = XMPUtil.readXMP(is);
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: No abbreviation

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 2]
```diff
@@ -515,6 +519,9 @@ private static XMPMetadata readRawXMP(InputStream inputStream) throws IOExceptio
         }
     }
 
+    /**
+     * @return null if no metadata has been found
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: Return optional instead?

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 3]
```diff
@@ -1233,6 +1243,18 @@ public static void main(String[] args) throws IOException,
     }
 
     /**
+     * see XMPUtil.hasMetadata(InputStream is)
+     */
+    public static boolean hasMetadata(Path p) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: p -> path

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java [hunk 4]
```diff
@@ -89,9 +93,10 @@ public void writeManually(File tempFile, String xmpString) throws Exception {
 
         try (PDDocument document = PDDocument.load(tempFile.getAbsoluteFile())) {
             if (document.isEncrypted()) {
-                System.err.println("Error: Cannot add metadata to encrypted document.");
-                //System.exit(1);
+                LOGGER.error("Cannot add metadata to encrypted document.");
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: Directly pass the string to the fail() method and remove Logger

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java [hunk 5]
```diff
@@ -232,8 +236,7 @@ public void setUp() throws IOException, COSVisitorException {
     @After
     public void tearDown() {
         if (!pdfFile.delete()) {
-            System.err
-                    .println("Note: Cannot delete temporary file (already deleted so the corresponding test passed).");
+            LOGGER.info("Note: Cannot delete temporary file (already deleted so the corresponding test passed).");
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: Better use TemporyFolder as described in the wiki https://github.com/JabRef/jabref/wiki/Code-Howtos#files-and-folders-in-tests

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java (hunk 2 above)
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: I knew that someone will complain :cry:

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java [hunk 6]
```diff
@@ -878,8 +876,8 @@ public void testReadWriteDC() throws IOException, TransformerException {
 
         try (PDDocument document = PDDocument.load(pdfFile.getAbsoluteFile())) {
             if (document.isEncrypted()) {
-                System.err.println("Error: Cannot add metadata to encrypted document.");
-                //System.exit(1);
+                LOGGER.error("Cannot add metadata to encrypted document.");
+                Assert.fail();
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: See above

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java [hunk 7]
```diff
@@ -944,8 +942,8 @@ public void testWriteSingleUpdatesDCAndInfo() throws IOException, TransformerExc
...
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: See abve

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java (hunk 5 above)
Comment (koppor) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: I'll replace it by [deleteOnExit](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#deleteOnExit%28%29).

**Comment by koppor** [comment]
Tried to address all comments and created different commits for different issues to ease reviewing - hopefully.

File: src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java (hunk 5 above)
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/xmp/XMPUtilTest.java]: As you prefer. The TemporyFolder rule has the advantage, that the temp-file is directly deleted after the test (not sure what happens if I stop the test process mid-way if you use deleteOnExit).

**Comment by tobiasdiez** [comment]
LGTM

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 8]
```diff
@@ -105,9 +113,11 @@ public static void writeXMP(String filename, BibEntry entry,
      *             than remove a lock or cancel the operation.
      */
     public static List<BibEntry> readXMP(File file) throws IOException {
-        try (FileInputStream is = new FileInputStream(file)) {
-            return XMPUtil.readXMP(is);
+        List<BibEntry> result = Collections.EMPTY_LIST;
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: Collections.empyList() is preferred

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 9]
```diff
@@ -1233,21 +1239,37 @@ public static void main(String[] args) throws IOException,
...
      * itself from the PDF.
      *
-     * @param is
-     *            The inputstream to read the PDF from.
+     * @param inputsStream
+     *            The inputStream to read the PDF from.
      * @return whether a BibEntry was found in the given PDF.
      */
-    public static boolean hasMetadata(InputStream is) {
+    public static boolean hasMetadata(InputStream inputsStream) {
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: inputStream instead of inputsStream

**Comment by koppor** [comment]
Fixed in eb26ff3.

**Pull request merged**